spec/models/customer_spec.rb

```diff
@@ -111,5 +111,21 @@
         expect(Customer.managed_by(guest)).to match_array []
       end
     end
+
+    context 'with_completed_orders' do
+      let!(:customer) { create(:customer) }
+      let!(:customer2) { create(:customer) }
+      let!(:customer3) { create(:customer) }
+      let!(:customer4) { create(:customer) }
+
+      let!(:order_ready_for_details) { create(:order_ready_for_details, customer: customer) }
+      let!(:order_ready_for_payment){ create(:order_ready_for_payment, customer: customer2) }
+      # let!(:order_ready_for_confirmation) { create(:order_ready_for_confirmation, state: :confirmation, customer: customer3) }
```

Comment:
When I used this, I expected to have an order with state = `confirmation`, but I got a `completed` order. is that ok?

Reply:
@jibees @filipefurtad0 any feedback on the order state question ☝️

Reply:
humm ... I don't think so. 
I've tested in `spec/system/consumer/split_checkout_spec.rb#836` and the order state is `confirmation` (I've printed the `order.state` via `pp` in the console).
I've tested with your branch: 
 - I think you don't need to set the state to `confirmation`
 - I've printed the order state around `order.next!` in the `order_ready_for_confirmation` factory. In your case, this transitions from `payment` to `complete` although it should transition from `payment` to `confirmation`. 
 - The `split_checkout` feature change the state machine for an order. So I've activate it in your case, since it is activated by default now. 
 - And it works! 🎉 

The diff:
```diff
diff --git a/spec/models/customer_spec.rb b/spec/models/customer_spec.rb
index f14cbb16f1..a8b737e8dd 100644
--- a/spec/models/customer_spec.rb
+++ b/spec/models/customer_spec.rb
@@ -112,19 +112,25 @@ describe Customer, type: :model do
       end
     end
 
-    context 'with_completed_orders' do
-      let!(:customer) { create(:customer) }
-      let!(:customer2) { create(:customer) }
-      let!(:customer3) { create(:customer) }
-      let!(:customer4) { create(:customer) }
-
-      let!(:order_ready_for_details) { create(:order_ready_for_details, customer: customer) }
-      let!(:order_ready_for_payment){ create(:order_ready_for_payment, customer: customer2) }
-      # let!(:order_ready_for_confirmation) { create(:order_ready_for_confirmation, state: :confirmation, customer: customer3) }
-      let!(:completed_order) {create(:completed_order_with_totals, customer: customer4)}
-
-      it 'returns customers with completed orders' do
-        expect(Customer.with_completed_orders).to match_array [customer4]
+    context "with split checkout enabled" do
+      before do
+        Flipper.enable(:split_checkout)
+      end
+
+      context 'with_completed_orders' do
+        let!(:customer) { create(:customer) }
+        let!(:customer2) { create(:customer) }
+        let!(:customer3) { create(:customer) }
+        let!(:customer4) { create(:customer) }
+
+        let!(:order_ready_for_details) { create(:order_ready_for_details, customer: customer) }
+        let!(:order_ready_for_payment){ create(:order_ready_for_payment, customer: customer2) }
+        let!(:order_ready_for_confirmation) { create(:order_ready_for_confirmation, customer: customer3) }
+        let!(:completed_order) { create(:completed_order_with_totals, customer: customer4) }
+
+        it 'returns customers with completed orders' do
+          expect(Customer.with_completed_orders).to match_array [customer4]
+        end
       end
     end
   end

``` 


**Thoughts**
We quickly need to either:
 - remove the legacy checkout and therefore feature toggle `split_checkout`
 - nor runs the specs with feature toggle `split_checkout` activated by default to maybe spot some others bugs?

Reply:
At this level, I'll just enable split_checkout on the context as you did.

I was thinking if we also need to include customers with orders having the state 'resumed'.
From what I understood, a resumed order will not transition to `completed`, right?